packages/repo/src/blockstore/memory-blockstore.ts

```diff
@@ -2,43 +2,63 @@ import { CID } from 'multiformats/cid'
 import IpldStore from './ipld-store'
 
 export class MemoryBlockstore extends IpldStore {
-  map: Map<string, Uint8Array>
+  blocks: Map<string, Uint8Array>
+  staged: Map<string, Uint8Array>
 
   constructor() {
     super()
-    this.map = new Map()
+    this.blocks = new Map()
+    this.staged = new Map()
   }
 
-  async getBytes(k: CID): Promise<Uint8Array> {
-    const v = this.map.get(k.toString())
-    if (!v) throw new Error(`Not found: ${k.toString()}`)
-    return v
+  async getBytes(cid: CID): Promise<Uint8Array> {
+    const fromStaged = this.staged.get(cid.toString())
+    if (fromStaged) return fromStaged
+    const fromBlocks = this.blocks.get(cid.toString())
+    if (fromBlocks) return fromBlocks
+    throw new Error(`Not found: ${cid.toString()}`)
   }
 
-  async putBytes(k: CID, v: Uint8Array): Promise<void> {
-    this.map.set(k.toString(), v)
+  // async putBytes(k: CID, v: Uint8Array): Promise<void> {
+  //   this.map.set(k.toString(), v)
+  // }
+
```

Comment:
```suggestion
```

Reply:
lol really did a bad job cleaning up comments on this one 😅 😅